### ISSUE TITLE
Crash with NoClassDefFoundError on Samsung devices

### ISSUE THREAD

**Issue by llweifall** [pockethub/PocketHub#997]
Dear developers of open-gpstracker,

I have found out that the method setSupportActionBar of android support library v7 is used in many activities . However, this method crashed the app on Samsung devices.
I found the discussions in the bug tracking system of Android: [here](https://code.google.com/p/android/issues/detail?id=78377). 
Numbers of apps are affected by this problem. And fixes were found effective by using the proguard like in comment [#308](https://code.google.com/p/android/issues/detail?id=78377#c308) or somebody reports that there's a valid workaround for this issue in 24.0.0-alpha1.

Thank you very much if you can fix this. Hope this information is helpful. :)

**Comment by hzsweers** [comment]
we are not open-gpstracker.

**Comment by llweifall** [comment]
I am sincerely sorry for the typo...I found the crash on both the two apps. So I mistakenly typed gpstracker. I wanted to report this bug in PocketHub.
Sorry again for my stupid mistake.

**Comment by hzsweers** [comment]
In either case, I don't really care to support bugged firmware versions from OEMs. PRs welcome, but not going to actively look into this

**Comment by llweifall** [comment]
Thank you very much. I may see if I can make a valid pull request.

**Comment by Meisolsson** [comment]
There's a ProGuard "fix" for this. Has worked well for me. Just adding this code to proguard-rules.pro

```
-keep class !android.support.v7.internal.view.menu.**,** {*;}
```

**Comment by llweifall** [comment]
@Meisolsson Thank you very much for your information. I observed similar fixes in other apps as well. And I also found that it seems when using a v7 library newer than 23.1.1 they seems to add `-keep class !android.support.v7.view.menu.**,** {*;}` instead of `android.support.v7.internal.view.menu.**,** {*;}`.
I wonder which one is proper.
And I wonder if this fix worth to be added in the project. I believe there are considerable users using Samsung devices who will also suffer from this problem.
Thanks again. :)
